networkit/cpp/graph/Graph.cpp

```diff
@@ -106,45 +107,68 @@ Graph::Graph(const Graph &G, bool weighted, bool directed, bool edgesIndexed)
         outEdges.resize(z);
+        if (weighted)
+            outEdgeWeights.resize(z);
+        if (G.hasEdgeIds() && edgesIndexed)
+            outEdgeIds.resize(z);
         for (node u = 0; u < z; u++) {
-
             // copy both out and in edges into our new outEdges
             outEdges[u].reserve(G.outEdges[u].size() + G.inEdges[u].size());
             outEdges[u].insert(outEdges[u].end(), G.outEdges[u].begin(), G.outEdges[u].end());
-            outEdges[u].insert(outEdges[u].end(), G.inEdges[u].begin(), G.inEdges[u].end());
-        }
-        if (weighted) {
-            if (G.isWeighted()) {
-                // same for weights
-                outEdgeWeights.resize(z);
-                for (node u = 0; u < z; u++) {
+            if (weighted) {
+                if (G.isWeighted()) {
+                    // same for weights
                     outEdgeWeights[u].reserve(G.outEdgeWeights[u].size()
                                               + G.inEdgeWeights[u].size());
                     outEdgeWeights[u].insert(outEdgeWeights[u].end(), G.outEdgeWeights[u].begin(),
                                              G.outEdgeWeights[u].end());
-                    outEdgeWeights[u].insert(outEdgeWeights[u].end(), G.inEdgeWeights[u].begin(),
-                                             G.inEdgeWeights[u].end());
-                }
-            } else {
-                // we are undirected, so no need to write anything into inEdgeWeights
-                outEdgeWeights.resize(z);
-                for (node u = 0; u < z; u++) {
+                } else {
+                    // we are undirected, so no need to write anything into inEdgeWeights
                     outEdgeWeights[u] =
                         std::vector<edgeweight>(outEdges[u].size(), defaultEdgeWeight);
                 }
             }
-        }
-        if (G.hasEdgeIds() && edgesIndexed) {
-            for (node u = 0; u < z; u++) {
-
+            if (G.hasEdgeIds() && edgesIndexed) {
                 // copy both out and in edges ids into our new outEdgesIds
                 outEdgeIds[u].reserve(G.outEdgeIds[u].size() + G.inEdgeIds[u].size());
                 outEdgeIds[u].insert(outEdgeIds[u].end(), G.outEdgeIds[u].begin(),
                                      G.outEdgeIds[u].end());
-                outEdgeIds[u].insert(outEdgeIds[u].end(), G.inEdgeIds[u].begin(),
-                                     G.inEdgeIds[u].end());
             }
         }
+        for (node u = 0; u < z; u++) {
+            // this is necessary to avoid multi edges, because both u -> v and v -> u can exist in G
+            count edgeSurplus = 0;
+            for (count i = 0; i < G.inEdges[u].size(); i++) {
+                node v = G.inEdges[u][i];
+                bool alreadyPresent = false;
+                for (count j = 0; j < G.outEdges[u].size(); j++) {
+                    if (v == G.outEdges[u][j]) { // the edge already exists as an out edge
```

Comment:
We can remove some indentations and make the code more readable
```suggestion
                    if (v != G.outEdges[u][j])
                        continue;
```